### ISSUE TITLE
Add check for None before creating a tmp dir

### ISSUE THREAD

**Issue by DobroAlex** [grindsa/acme2certifier#47]
I'm using example/ca_handler/cmp_ca_handler.py as a ca_handler and example/db_handler/wsgi_handler.py as a db_handler; my acme/acme_srv.cfg is 
```
[DEFAULT]
debug: False

[Nonce]
# disable nonce check. THIS IS A SEVERE SECURTIY ISSUE! Please do only for testing/debugging purposes
nonce_check_disable: False

[CAhandler]
# CA specific options
api_host: "https://acme-v02.api.letsencrypt.org/directory"

[Certificate]
revocation_reason_check_disable: False

[Challenge]
# when true disable challenge validation. Challenge will be set to 'valid' without further checking
# THIS IS A SEVERE SECURTIY ISSUE! Please do only for testing/debugging purposes
challenge_validation_disable: False

[Order]
tnauthlist_support: False
```

So in the used ca_handler  on lines 186-188 in the function load_config() there are strings:
```
# create temp dir if needed
if not os.path.exists(self.tmp_dir):
   os.makedirs(self.tmp_dir)
```

Those may throw a TypeError if self.tmp_dir is None. I think it should be something like:
```
if self.tmp_dir is not  None and  if not os.path.exists(self.tmp_dir):
   os.makedirs(self.tmp_dir)
```

**Comment by DobroAlex** [comment]
In my case, I'm using certbot with certbot nginx and it actually throws this exception

**Comment by DobroAlex** [comment]
The commands I'm passing to certbot (maybe there is something wrong with them):
```
"certonly", "--nginx", "--noninteractive", "-d icanbreakit.xyz", "-m some@gmail.com", "--agree-tos", "--server", "http://127.0.0.1:8088"
```
(This app is running on my the local machine, yes, the domain is hosted here too)

**Comment by grindsa** [comment]
Thank you for reporting. You forgot to specify the `cmp_tmp_dir`  parameter in `acme_srv.cfg`.  Check the [handler documentation](https://github.com/grindsa/acme2certifier/blob/master/docs/cmp.md)  for further information. I pushed a fix aa9551af into a [cmpv2_handler error branch](https://github.com/grindsa/acme2certifier/tree/cmpv2_handler_error) but I could not verify it as my cmpv2 server is down. Will further look into it in the upcoming days.

BTW: the cmp handler has originally been developed for python 2 and not been updated for quite some time. So there might be other issues coming up.

**Comment by grindsa** [comment]
I pushed a couple of fixes into [0.11.1](https://github.com/grindsa/acme2certifier/releases/tag/0.11.1). Give it a try and let me know if it works for you.

**Comment by DobroAlex** [comment]
Thanks, that's better
